tests/storage/__init__.py

```diff
@@ -1,98 +1,88 @@
 import os
 
 import pytest
+from ert_shared import ERT_STORAGE
 from ert_shared.storage.blob_api import BlobApi
 from ert_shared.storage.model import Blobs, Entities
 from ert_shared.storage.rdb_api import RdbApi
+from ert_shared.storage.storage_api import StorageApi
 from sqlalchemy import create_engine
 from sqlalchemy.orm import Session
 
 
-@pytest.fixture(scope="session")
-def engine():
-    engine = create_engine("sqlite:///:memory:", echo=False)
-    engine.execute("pragma foreign_keys=on")
-    return engine
-
+@pytest.fixture
+def initialize_databases(tmpdir):
+    rdb_url = f"sqlite:///{tmpdir}/entities.db"
+    blob_url = f"sqlite:///{tmpdir}/blobs.db"
 
-@pytest.fixture(scope="session")
-def tables(engine):
-    Entities.metadata.create_all(engine)
-    Blobs.metadata.create_all(engine)
-    yield
-    Entities.metadata.drop_all(engine)
-    Blobs.metadata.drop_all(engine)
+    ERT_STORAGE.initialize(rdb_url=rdb_url, blob_url=blob_url)
 
 
 @pytest.fixture
-def db_connection(engine, tables):
-    """Returns an sqlalchemy session, and after the test tears down everything properly."""
-    connection = engine.connect()
-    transaction = connection.begin()
-
-    yield connection
+def apis(initialize_databases):
+    rdb_session = ERT_STORAGE.RdbSession()
+    blob_session = ERT_STORAGE.BlobSession()
 
-    transaction.rollback()
-    connection.close()
+    rdb_api = RdbApi(session=rdb_session)
+    blob_api = BlobApi(session=blob_session)
 
+    try:
+        yield (rdb_api, blob_api)
 
-@pytest.fixture(scope="module")
-def db_info(tmp_path_factory):
-    def add_blob(data):
-        ret = blob.add_blob(data)
-        blob.flush()
-        return ret.id
+        rdb_session.commit()
```

Comment:
We should rollback always during tests. Otherwise we'll end up with tests depending on other tests.

```python3
try:
    yield rdb_api, blob_api
finally:
    rdb_session.rollback()
    blob_session.rollback()
    rdb_session.close()
    blob_session.close()
```

Reply:
yes - if i fix your comment below ⬇️  that would impact this ⬆️ . Good catch ... getting kind of blind when doing so much refactoring trying while trying to minimize the amount of  changes at the same time